Add a consistency check for the model editor

koesie10 opened the pull request:
This adds an internal consistency check for the model editor. This catches cases where the model editor thinks something is supported, but the model editor queries return that it it not supported.

This would for example have caught https://github.com/github/codeql/pull/15089 without needing to manually check the query.

![Screenshot 2023-12-14 at 12 09 40](https://github.com/github/vscode-codeql/assets/1112623/903964d7-05f2-42af-a48a-cf7f3b5c8649)

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

charisk reviewed (approved):
LGTM!

I wouldn't even hide it behind a flag since it's only logging to the extension log. Those logs could be useful when debugging a user problem.

koesie10 commented:
> I wouldn't even hide it behind a flag since it's only logging to the extension log. Those logs could be useful when debugging a user problem.

Good point, I've removed the feature flag. I ran into some incorrect checks for the Ruby model editor, so I've excluded `type` models from being considered as supported. I've also updated the message to reference the model editor since it could now be shown to users if there is a bug in the queries or the model editor.

charisk reviewed (approved):
🚢
